src/hooks/useInput.js

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useRef, useState } from 'react'
+import { useFocusEnter, useFocusLeave } from '@aragon/ui'
+
+export function useInput(validationFn) {
+  const [value, setValue] = useState('')
+  const [status, setStatus] = useState('empty')
+
+  const inputRef = useRef()
+
+  const handleBlur = useCallback(() => {
+    const invalidValue = !validationFn(value)
```

Comment:
Doing this is kind of weird, what do you think about something like

```
setStatus(validationFn(value) ? 'valid' : 'invalid')
```

Instead? The main part I don't like about the above is `!validationFn(value)`—this is easy to miss and the variable separation makes it harder to think about what state the validation function will drive.